src/components/ProfilePopout/ProfilePopout.jsx

```diff
@@ -21,10 +21,23 @@ const ProfilePopout = () => {
   const hideProfilePopout = () => {
     ref.current.style.display = 'none';
     dispatch(setCloseProfilePopout(true));
+    localStorage?.setItem('hideProfilePopout', 'true');
   };
+  useEffect(() => {
+    if (localStorage?.getItem('hideProfilePopout')) {
+      hideProfilePopout();
+    }
+  }, []);
 
   return (
-    <ProfilePopoutContainer ref={ref} status={status}>
+    <ProfilePopoutContainer
```

Comment:
you don't need ref to handle the popup and not the best approach to handle this. You are first rendering the popup and then hiding it which is kind of wrong.

Instead, check if `hideProfilePopout` exists in local storage. Based on results,

1. if `false`, you show the popup. And when user closes the popup, set `hideProfilePopout` to true.
2. if `true`, you don't show it. You don't need a ref to hide it. Just use &`&` operator

Reply:
I understand this.
But I did not introduce a ref in this code. It was done by @Ishagoyal. I don't know why she did that, but I decided that I should not refactor this...

Reply:
> you don't need ref to handle the popup and not the best approach to handle this. You are first rendering the popup and then hiding it which is kind of wrong.
> 
> Instead, check if `hideProfilePopout` exists in local storage. Based on results,
> 
> 1. if `false`, you show the popup. And when user closes the popup, set `hideProfilePopout` to true.
> 2. if `true`, you don't show it. You don't need a ref to hide it. Just use &`&` operator

Thank you, I have removed unnecessary ref and also did a small refactoring of my latest code.